core/src/main/java/tech/pegasys/eth2signer/core/multikey/metadata/SigningMetadataFile.java

```diff
@@ -19,19 +19,19 @@
 
   protected String baseFilename;
 
-  public SigningMetadataFile(final String filename) {
-    this.baseFilename = getFilenameWithoutExtension(filename);
+  public SigningMetadataFile(final String filename, final String fileExtension) {
```

Comment:
smell: maybe filename should be renamed basename? This class seems to do validation that the input data was correct .. which, feels like you're expecting people to use it incorrectly - which should be hard to do.

Maybe it should be constructed with a Path - then use org.apache.commons.io.FilenameUtils?

Reply:
done